homeassistant/components/media_player/clementine.py

```diff
@@ -0,0 +1,225 @@
+"""
+Support for Clementine Music Player as media player.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.clementine/
+"""
+
+import time
+from datetime import timedelta
+import logging
+import asyncio
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.media_player import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, PLATFORM_SCHEMA,
+    SUPPORT_VOLUME_STEP, SUPPORT_SELECT_SOURCE, SUPPORT_PLAY, MEDIA_TYPE_MUSIC,
+    SUPPORT_VOLUME_SET, MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PORT, CONF_ACCESS_TOKEN,
+    STATE_OFF, STATE_PLAYING, STATE_PAUSED, STATE_UNKNOWN)
+import homeassistant.util as util
+
+REQUIREMENTS = ['python-clementine-remote==1.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Clementine Remote'
+
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+SUPPORT_CLEMENTINE = SUPPORT_PAUSE | SUPPORT_VOLUME_STEP | \
+                     SUPPORT_PREVIOUS_TRACK | SUPPORT_VOLUME_SET | \
+                     SUPPORT_NEXT_TRACK | \
+                     SUPPORT_SELECT_SOURCE | SUPPORT_PLAY
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=5500): cv.positive_int,
+    vol.Optional(CONF_ACCESS_TOKEN, default=None): cv.positive_int,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Clementine platform."""
+    from clementineremote import ClementineRemote
+    client = ClementineRemote(config.get(CONF_HOST), config.get(CONF_PORT),
+                              config.get(CONF_ACCESS_TOKEN), reconnect=True)
+
+    add_devices([ClementineDevice(client, config[CONF_NAME])])
+
+
+class ClementineDevice(MediaPlayerDevice):
+    """Representation of Clementine Player."""
+
+    def __init__(self, client, name):
+        """Initialize the Clementine device."""
+        self._client = client
+        self._name = name
+        self._muted = False
+        self._volume = 0.0
+        self._track_id = 0
+        self._last_track_id = 0
+        self._track_name = ''
+        self._track_artist = ''
+        self._track_album_name = ''
+        self._state = STATE_UNKNOWN
+
+        self.update()
+
+    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update(self):
```

Comment:
Understood. I wonder if there is a lower level introduction I missed. I'll go and review the docs anyway. I have used a smaller interval, because updating this component does not involve any network access.